library/ssl_tls13_keys.c

```diff
@@ -1202,7 +1202,15 @@ int mbedtls_ssl_tls13_generate_handshake_keys( mbedtls_ssl_context *ssl,
     unsigned char transcript[MBEDTLS_TLS1_3_MD_MAX_SIZE];
     size_t transcript_len;
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
```

Comment:
The TLS 1.3 code requires PSA crypto: it effectively has USE_PSA_CRYPTO always on. So in TLS 1.3 code, there should be no check for `MBEDTLS_USE_PSA_CRYPTO`, and eventually only PSA code paths. Some TLS 1.3 code was originally written for the classic API; when migrating it, just switch to PSA without leaving a legacy path.

Reply:
There's one exception: `mbedtls_ssl_tls13_populate_transform()`, see this comment: https://github.com/Mbed-TLS/mbedtls/issues/5797#issuecomment-1127356552

Reply:
But indeed here and in `mbedtls_ssl_tls13_generate_application_keys()` I agree we should only have the PSA version, good catch!